Examples/test-suite/perl5/rename_qualified_runme.pl

```diff
@@ -0,0 +1,36 @@
+use strict;
+use warnings;
+use Test::More tests => 12;
+BEGIN { use_ok('rename_qualified') }
+require_ok 'rename_qualified';
+
+sub try (&$$) {
+  my($test, $want, $name) = @_;
+  my $tb = Test::Builder->new();
+  my $got = eval { $test->() };
+  if($@) {
+    my $err = $@;
+    $err =~ s/^/    /mg;
+    $tb->diag($err);
+    return $tb->ok(0, $name);
+  }
+  return $tb->is_eq($got, $want, $name);
+}
+
+TODO: {
+  local $TODO = q(need to diagnose "Can't wrap XXX unless renamed to a valid identifier.");
```

Comment:
This warning message happens because you can't have colons in a symbol name. The %rename should then be ignored. If it is being accepted, it is a bug.

I think you are trying to use %rename to put symbols into an additional Perl package namespace. This is an abuse of %rename and should not be allowed. I don't know what the package support is like in SWIG for Perl, but should probably be used instead. We don't have a mechanism in SWIG for moving symbols into a different target namespace equivalent (like a Perl package). That is a whole new dimension of support to be added if deemed useful (I don't think it is). Perl should support %nspace first.

Can you please modify the patch to test the fix without this unsupported use of %rename?

Reply:
You're right, I am trying to use %rename to move related groups of functionality into separate packages.  It sounds like Perl handling needs to be more aggressive about checking valid identifiers than it currently is to prevent this abuse.  As evidenced by the one case I found working, constants need some additional validation.  I'll work out what needs to be done there.